src/buttonMaker/IconForm.jsx

```diff
@@ -0,0 +1,98 @@
+import * as allIcons from "@chakra-ui/icons";
+import "./buttonMaker.css";
+import { changeForm, clickForm } from "./formLogic";
+
+// A module of all the availabe icons
+let { Icon, createIcon, ...iconsModule } = allIcons;
+
+// A list of all the icon names
+let iconsName = Object.keys(iconsModule).map((e) => e);
+
+// A dropdown for the user to pick their icon
+function DropdownItem({ iconName, Icon, formDispatch }) {
+    return (
+        <button
+            name={"icon"}
+            id={iconName.replace(/\s/g, "")}
+            onClick={(e) => clickForm(e, formDispatch)}
+        >
+            <Icon /> &nbsp;
+            {iconName}
+        </button>
+    );
+}
+
+// A function to make the icon names more pleasant to read
+function pascalToSpaces(expression) {
+    return (
+        expression
+            // Look for lower-case letters followed by upper-case letters
+            .replace(/([a-z\d])([A-Z])/g, "$1 $2")
+            // Look for lower-case letters followed by numbers
+            .replace(/([a-zA-Z])(\d)/g, "$1 $2")
+            .replace(/^./, function (str) {
+                return str.toUpperCase();
+            })
+            // Remove any white space left around the word
+            .trim()
+    );
+}
+
+function IconForm({ formState, formDispatch }) {
+    // Making an object of the module of the icons
```

Comment:
praise: great job with this section! It's quite advanced, and you've done it well!